### PR TITLE
Add documentation for Snaps lifecycle hooks

### LINKED ISSUE #1076: Document lifecycle-hooks for Snaps
Add the endowment to the endowments page and add the handlers to the Snaps exports. 

Optional: add a feature card on the Snaps introduction for lifecycle hooks.

### PR DESCRIPTION
**Pull request by Montoya**
This closes #1076

### CODE REVIEW

File: snaps/reference/exports.md [hunk 1]
```diff
@@ -270,3 +270,143 @@ module.exports.onCronjob = async ({ request }) => {
...
+[`endowment:lifecycle-hooks`](permissions.md#endowmentlifecycle-hooks) permission.
+:::
+
+#### Parameters
+
+None.
+
+#### Returns
+
+A promise containing the return of the implemented method.
```
Comment (FrederikBolding) [inline snaps/reference/exports.md:290]:
```suggestion
```

I don't think mentioning a return value makes sense here, similar to cronjobs.

File: snaps/reference/exports.md [hunk 2]
```diff
@@ -270,3 +270,143 @@ module.exports.onCronjob = async ({ request }) => {
...
```
Comment (FrederikBolding) [inline snaps/reference/exports.md:357]:
```suggestion
```

```suggestion
```

I don't think mentioning a return value makes sense here, similar to cronjobs.

File: snaps/reference/exports.md (hunk 1 above)
Comment (Montoya) [reply snaps/reference/exports.md:290]: I think it's worthwhile to mention that it returns a promise

**Review by FrederikBolding — APPROVED** [review]

File: snaps/reference/permissions.md [hunk 3]
```diff
@@ -101,6 +101,22 @@ Specify this permission in the manifest file as follows:
 }
 ```
 
+### endowment:lifecycle-hooks
+
+To run an action when the user installs or updates the Snap, a Snap must request the `endowment:lifecycle-hooks` permission.
+This permission allows the Snap to export the 
+[`onInstall`](../reference/exports.md#oninstall) and 
+[`onUpdate`](../reference/exports.md#onupdate) 
+methods, which will be called by MetaMask after a successful install or update, respectively.
```
Comment (alexandratran) [inline snaps/reference/permissions.md:110]:
```suggestion
methods, which MetaMask calls after a successful installation or update, respectively.
```

**Review by alexandratran — APPROVED** [review]
LGTM

**Pull request merged**